### PR TITLE
Fix dataKey accessor for columns

### PR DESCRIPTION
**Pull request by scaredcat**
* When displaying multiple columns under a single value, only display first header
* Default comparison options for remoteChoiceFilter
* Boolean filter only returns value to onChange instead of an object

### CODE REVIEW

File: src/utils/proptypes.js [hunk 1]
```diff
@@ -8,21 +8,29 @@ const STATUS_CHOICES = {
...
-  if (typeof value === 'object') {
-    validObject = Object.entries(value).every(
-      ([key, value]) => (typeof key === 'number' || 'string') && typeof value === 'boolean')
-  } else {
-    validObject = false
-  }
-
-  if (value !== ALL_SELECTED && !validObject) {
-    return new Error(`Invalid prop \`${propName}\` of \`${value}\`
+function selectionTypeValidator(isRequired) {
```
Comment (tabby-or-not) [inline src/utils/proptypes.js]: Nice!

File: src/filters/types/__tests__/__snapshots__/Boolean.test.js.snap [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 exports[`Boolean Snapshots has custom labels 1`] = `
 <Choice
-  onChange={[MockFunction]}
+  onChange={[Function]}
```
Comment (tabby-or-not) [inline src/filters/types/__tests__/__snapshots__/Boolean.test.js.snap]: Have we changed jest versions?

File: src/data-renderers/utils/__tests__/functions.test.js [hunk 3]
```diff
@@ -10,10 +10,7 @@ describe('getValueFromAccessor', () => {
   it('returns value for a nested row/dotted key', () => {
     expect(getValueFromAccessor({a: {b: 'c'}}, 'a.b')).toBe('c')
   })
-  it('returns value up to point for incorrect key', () => {
-    expect(getValueFromAccessor({a: {b: 'c'}}, 'a.c')).toEqual({b: 'c'})
-  })
-  it('returns value without repetitive weirdness for incorrect key', () => {
-    expect(getValueFromAccessor({a: {b: 'c'}}, 'a.c.b')).toEqual({b: 'c'})
+  it('returns null for incorrect key', () => {
```
Comment (tabby-or-not) [inline src/data-renderers/utils/__tests__/functions.test.js]: What's the reason for this change?

Comment (scaredcat) [reply src/data-renderers/utils/__tests__/functions.test.js]: sometimes there might be in intermediate value returned from the current data when new data is loaded. eg
```javascript
{ attributes: {one: 'some value'} }
/// goes to
{ attributes: {one: 'some value', two: 'some other value'} }
```
when displaying the second column the `value` passed to it from an dataKey of `attributes.two` would be the entire attributes object as `.two` does not yet exist.

I feel this just makes it a bit cleaner

File: src/filters/types/__tests__/__snapshots__/Boolean.test.js.snap (hunk 2 above)
Comment (scaredcat) [reply src/filters/types/__tests__/__snapshots__/Boolean.test.js.snap]: the difference here is that `onChange` is now a function within the `Boolean` react class. This now calls the passed in `mock function props` instead of accessing them directly
